test/integration/controllermanager/namespacedcloudprofile/namespacedcloudprofile_test.go

```diff
@@ -0,0 +1,489 @@
+		var (
+			parentCloudProfile     *gardencorev1beta1.CloudProfile
+			namespacedCloudProfile *gardencorev1beta1.NamespacedCloudProfile
+			shoot                  *gardencorev1beta1.Shoot
+		)
+
+		BeforeEach(func() {
+			parentCloudProfile = (&parentCloudProfileBase).DeepCopy()
+			namespacedCloudProfile = (&namespacedCloudProfileBase).DeepCopy()
+
+			shoot = &gardencorev1beta1.Shoot{
+				ObjectMeta: metav1.ObjectMeta{
+					GenerateName: testID + "-",
+					Namespace:    testNamespace.Name,
+				},
+				Spec: gardencorev1beta1.ShootSpec{
+					SecretBindingName: ptr.To("my-provider-account"),
+					Region:            "foo-region",
+					Provider: gardencorev1beta1.Provider{
+						Type: "aws",
+						Workers: []gardencorev1beta1.Worker{
+							{
+								Name:    "cpu-worker",
+								Minimum: 2,
+								Maximum: 2,
+								Machine: gardencorev1beta1.Machine{Type: "large"},
+							},
+						},
+					},
+					Kubernetes: gardencorev1beta1.Kubernetes{Version: "1.26.1"},
+					Networking: &gardencorev1beta1.Networking{Type: ptr.To("foo-networking")},
+				},
+			}
+		})
+
+		JustBeforeEach(func() {
+			By("Create parent CloudProfile")
+			Expect(testClient.Create(ctx, parentCloudProfile)).To(Succeed())
+			log.Info("Created parent CloudProfile for test", "parentCloudProfile", client.ObjectKeyFromObject(parentCloudProfile))
+
+			By("Create NamespacedCloudProfile")
+			namespacedCloudProfile.Spec.Parent = gardencorev1beta1.CloudProfileReference{
+				Kind: "CloudProfile",
+				Name: parentCloudProfile.Name,
+			}
+			Expect(testClient.Create(ctx, namespacedCloudProfile)).To(Succeed())
+			log.Info("Created NamespacedCloudProfile for test", "namespacedCloudProfile", client.ObjectKeyFromObject(namespacedCloudProfile))
+
+			if shoot != nil {
+				By("Create Shoot")
+				shoot.Spec.CloudProfile = &gardencorev1beta1.CloudProfileReference{
+					Kind: "NamespacedCloudProfile",
+					Name: namespacedCloudProfile.Name,
+				}
+				Expect(testClient.Create(ctx, shoot)).To(Succeed())
+				log.Info("Created shoot for test", "shoot", client.ObjectKeyFromObject(shoot))
+
+				By("Wait until manager has observed Shoot")
+				// Use the manager's cache to ensure it has observed the Shoot.
+				// Otherwise, the controller might clean up the NamespacedCloudProfile too early because it thinks all referencing
+				// Shoots are gone. Similar to https://github.com/gardener/gardener/issues/6486 and
+				// https://github.com/gardener/gardener/issues/6607.
+				Eventually(func() error {
+					return mgrClient.Get(ctx, client.ObjectKeyFromObject(shoot), &gardencorev1beta1.Shoot{})
+				}).Should(Succeed())
+
+				DeferCleanup(func() {
+					By("Delete Shoot")
+					Expect(client.IgnoreNotFound(testClient.Delete(ctx, shoot))).To(Succeed())
+				})
+			}
+
+			DeferCleanup(func() {
+				By("Delete NamespacedCloudProfile")
+				Expect(client.IgnoreNotFound(testClient.Delete(ctx, namespacedCloudProfile))).To(Succeed())
```

Comment:
We don't expect a NotFound error here, so we should probably not ignore it to be on the safe side. Same for remaining occurrences in this file.